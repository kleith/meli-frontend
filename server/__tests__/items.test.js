var request = require('supertest');
var app = require('../app');

describe('Test /api/items request from an item id', function() {
  it('should get item information', async function(done) {
    var res = await request(app)
      .get('/api/items/MLA821329221')
      .expect('Content-Type', /json/);
    
    expect(res.statusCode).toEqual(200);
    expect(res.body).toHaveProperty('author');
    expect(res.body).toHaveProperty('item');
    done();
  });

  it('should get snpashot item information', async function(done) {
    var res = await request(app)
      .get('/api/items/MLA821329221')
      .expect('Content-Type', /json/);
    
    expect(res.body).toMatchSnapshot({
      item: expect.any(Object)
    });
    done();
  });

  it('should get an error', async function(done) {
    var res = await request(app)
      .get('/api/items/sdz.MLA821329221')
      .expect('Content-Type', /json/);
    
    expect(res.statusCode).toEqual(404);
    expect(res.body).toHaveProperty('message');
    done();
  });

  it('should get a snapshot error', async function(done) {
    var res = await request(app)
      .get('/api/items/sdz.MLA821329221')
      .expect('Content-Type', /json/);
    
    expect(res.body).toMatchSnapshot();
    done();
  });
});

describe('Test /api/items request from query search', function() {
  it('should get item\'s list', async function(done) {
    var res = await request(app)
      .get('/api/items?q=zapatillas')
      .expect('Content-Type', /json/);
    
    expect(res.statusCode).toEqual(200);
    expect(res.body).toHaveProperty('author');
    expect(res.body).toHaveProperty('categories');
    expect(res.body).toHaveProperty('items');
    done();
  });

  it('should get item\'s list snpashot results', async function(done) {
    var res = await request(app)
      .get('/api/items?q=zapatillas')
      .expect('Content-Type', /json/);
    
    expect(res.body).toMatchSnapshot({
      items: expect.any(Array)
    });
    done();
  });
});