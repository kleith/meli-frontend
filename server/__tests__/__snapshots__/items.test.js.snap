// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test /api/items request from an item id should get a snapshot error 1`] = `
Object {
  "cause": Array [],
  "error": "not_found",
  "message": "Item with id sdz.MLA821329221 not found.",
  "status": 404,
}
`;

exports[`Test /api/items request from an item id should get snpashot item information 1`] = `
Object {
  "author": Object {
    "lastname": "Martínez",
    "name": "Sergio",
  },
  "item": Any<Object>,
}
`;

exports[`Test /api/items request from query search should get item's list snpashot results 1`] = `
Object {
  "author": Object {
    "lastname": "Martínez",
    "name": "Sergio",
  },
  "categories": Array [
    "Ropa y Accesorios",
    "Calzado",
    "Zapatillas",
  ],
  "items": Any<Array>,
}
`;
